docarray/document/mixins/plot.py

```diff
@@ -96,15 +96,14 @@ def display(self, from_: Optional[str] = None):
 
     def _is_3d(self) -> bool:
         """
-        Tells if Document stores a 3D object saved as point cloud or vertices and face.
+        Tells if Document stores a 3D object saved as point cloud, RGB-D image or vertices and faces.
         :return: bool.
         """
         if self.uri and self.uri.endswith(tuple(Mesh.FILE_EXTENSIONS)):
             return True
-        elif (
-            self.tensor is not None
-            and self.tensor.shape[1] == 3
-            and self.tensor.ndim == 2
+        elif self.tensor is not None and (
+            (self.tensor.shape[-1] == 3 and self.tensor.ndim == 2)
+            or (self.tensor.shape[-1] == 4 and self.tensor.ndim == 3)
```

Comment:
lets dispactch this _is_3d to three function , is_3d_point_cloud, is_3d_vertices_and ....so that we don't duplicate the check in the display